system/Config/Services.php

```diff
@@ -626,14 +627,16 @@ public static function router(?RouteCollectionInterface $routes = null, ?Request
      * secure, most notably the CSRF protection tools.
      *
      * @return Security
+     *
+     * @TODO replace the first parameter type `?App` with `?SecurityConfig`
```

Comment:
If we change it, the existing following code would break:
```php
Services::security($config);
```
It is a breaking change.

Reply:
Don't we have any other choice? If someone will use an App config this service will not work correctly, due to a lack of/incorrect config variables.

Reply:
What do you mean? The first parameter type is `Config\App`, not `SecurityConfig`.

Reply:
> If we change it, the existing following code would break:
> 
> ```
> Services::security($config);
> ```
> 
> It is a breaking change.

It will not work anyway, since the lack of config variables. Therefore I see no point in leaving `Config\App` here.

Reply:
If there is code like this:
```php
        $config = config(App::class);
        Services::security($config);
```
If we change the first parameter to `?SecurityConfig`,
```
TypeError
CodeIgniter\Config\Services::security(): Argument #1 ($config) must be of type ?Config\Security, Config\App given
```
Am I misunderstanding something?

Reply:
Do we have any config variables in `ConfigApp` that the current Security class uses?

It seems to me that the names of the variables used in the `ConfigApp` are completely different.

* ConfigApp: `CSRFTokenName`, `CSRFHeaderName`...
* SecurityConfig: `tokenName`, `headerName`...

Reply:
This PR will remove all config variables in `Config\App` that the Security class used.
And the Security class will not use any config variable in `Config\App`.

And the first parameter type `?App` does not make sense. Because it is no longer used in the `security()` method. The `$config` is reassigned:

```php
$config = config(SecurityConfig::class);
```
But if we change the type, it may cause TypeError.

Do you say it is better to change it to `?SecurityConfig` now?

Reply:
> Do you say it is better to change it to `?SecurityConfig` now?

Yes, I think so. Allowing `Config\App` may cause some problems.

Reply:
Changed: https://github.com/codeigniter4/CodeIgniter4/pull/7630/commits/5adf1c4e55966a3342e0b702e0ddf371ecc5ef1c